x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -584,9 +590,17 @@ protected QueryTranslation asQuery(In in, boolean onAggs) {
             else {
                 Query q = null;
                 if (in.value() instanceof FieldAttribute) {
-                    FieldAttribute fa = (FieldAttribute) in.value();
                     // equality should always be against an exact match (which is important for strings)
-                    q = new TermsQuery(in.source(), fa.exactAttribute().name(), in.list());
+                    FieldAttribute fa = (FieldAttribute) in.value();
+                    List<Expression> list = in.list();
+                    // TODO: this needs to be handled inside the optimizer
```

Comment:
@matriv This case of eliminating nulls need to be moved from the query to the optimizer (maybe in a dedicated rule).
It seems to be inconsistent with the foldable method - currently `In` has a fairly scattered behavior when it comes to list that would be good to centralize.

Reply:
What was the issue? which tests were failing?

The In uses a HashSet to eliminate duplicates but if there are nulls in the list of values it keeps at list one, in order to implement the following behavior (adopted from Postegres):
```
postgres=# select 1 IN (2, 3, null);
 ?column?
----------

(1 row)

postgres=# select 1 IN (1, 2, 3, null);
 ?column?
----------
 t
(1 row)

postgres=# select 1 IN (2, 3);
 ?column?
----------
 f
(1 row)
```

Reply:
Eliminating the `null` from the list of values before doing the translation. This logic should be externalized from the query translation which should only worry about that not do any clean-up.